src/lib/services/client-metrics/metrics-service-v2.ts

```diff
@@ -62,30 +67,35 @@ export default class ClientMetricsServiceV2 {
         clientIp: string,
     ): Promise<void> {
         const value = await clientMetricsSchema.validateAsync(data);
-        const toggleNames = Object.keys(value.bucket.toggles);
-        if (toggleNames.length > 0) {
-            await this.featureToggleStore.setLastSeen(toggleNames);
-        }
+        const toggleNames = Object.keys(value.bucket.toggles).filter(
+            (name) =>
+                !(
+                    value.bucket.toggles[name].yes === 0 &&
+                    value.bucket.toggles[name].no === 0
+                ),
+        );
 
         this.logger.debug(`got metrics from ${clientIp}`);
 
-        const clientMetrics: IClientMetricsEnv[] = toggleNames
-            .map((name) => ({
-                featureName: name,
-                appName: value.appName,
-                environment: value.environment,
-                timestamp: value.bucket.start, //we might need to approximate between start/stop...
-                yes: value.bucket.toggles[name].yes,
-                no: value.bucket.toggles[name].no,
-            }))
-            .filter((item) => !(item.yes === 0 && item.no === 0));
+        const clientMetrics: IClientMetricsEnv[] = toggleNames.map((name) => ({
+            featureName: name,
+            appName: value.appName,
+            environment: value.environment,
+            timestamp: value.bucket.start, //we might need to approximate between start/stop...
+            yes: value.bucket.toggles[name].yes,
+            no: value.bucket.toggles[name].no,
+        }));
 
         if (this.config.flagResolver.isEnabled('batchMetrics')) {
             this.unsavedMetrics = collapseHourlyMetrics([
                 ...this.unsavedMetrics,
                 ...clientMetrics,
             ]);
+            this.lastSeenService.updateLastSeen(clientMetrics);
```

Comment:
Should this just be the default?

Reply:
I prefer to have it behind a flag, and the batch flag seems natural.